griptape/drivers/image_generation_model/stable_diffusion_3_pipeline_image_generation_model_driver.py

```diff
@@ -0,0 +1,94 @@
+
+if TYPE_CHECKING:
+    import torch
+    from PIL.Image import Image
+
+
+@define
+class StableDiffusion3PipelineImageGenerationModelDriver(BaseDiffusionPipelineImageGenerationModelDriver):
+    """Image generation model driver for Stable Diffusion 3 models.
+
+    For more information, see the HuggingFace documentation for the StableDiffusion3Pipeline:
+        https://huggingface.co/docs/diffusers/main/en/api/pipelines/stable_diffusion/stable_diffusion_3
+
+    Attributes:
+        width: The width of the generated image. Defaults to 1024. Must be a multiple of 64.
+        height: The height of the generated image. Defaults to 1024. Must be a multiple of 64.
+        seed: The random seed to use for image generation. If not provided, a random seed will be used.
+        guidance_scale: The strength of the guidance loss. If not provided, the default value will be used.
+        steps: The number of inference steps to use in image generation. If not provided, the default value will be used.
+        torch_dtype: The torch data type to use for image generation. If not provided, the default value will be used.
+    """
+
+    width: int = field(default=1024, kw_only=True, metadata={"serializable": True})
+    height: int = field(default=1024, kw_only=True, metadata={"serializable": True})
+    seed: Optional[int] = field(default=None, kw_only=True, metadata={"serializable": True})
+    guidance_scale: Optional[float] = field(default=None, kw_only=True, metadata={"serializable": True})
+    steps: Optional[int] = field(default=None, kw_only=True, metadata={"serializable": True})
+    torch_dtype: Optional[torch.dtype] = field(default=None, kw_only=True, metadata={"serializable": True})
+
+    def prepare_pipeline(self, model: str, device: Optional[str]) -> Any:
+        pipeline_params = {}
+        if self.torch_dtype is not None:
+            pipeline_params["torch_dtype"] = self.torch_dtype
+
+        # A model can be provided either as a path to a local file
+        # or as a HuggingFace model repo name.
+        sd3_pipeline = import_optional_dependency(
+            "diffusers.pipelines.stable_diffusion_3.pipeline_stable_diffusion_3"
+        ).StableDiffusion3Pipeline
+        if os.path.isfile(model):
+            # If the model provided is a local file (not a directory),
+            # we load it using the from_single_file method.
+            pipeline = sd3_pipeline.from_single_file(model, **pipeline_params)
+        else:
+            # If the model is a local directory or hosted on HuggingFace,
+            # we load it using the from_pretrained method.
+            pipeline = sd3_pipeline.from_pretrained(model, **pipeline_params)
+
+        # Move inference to particular device if requested.
+        if device is not None:
+            pipeline.to(device)
+
+        return pipeline
+
+    def make_image_param(self, image: Optional[Image]) -> Optional[dict[str, Image]]:
+        return None
```

Comment:
I guess this is why image is optional, but this feels odd.

Reply:
This is odd, and a half-measure. Zooming out, I think the time has come to admit the design where each driver implements a suite of common image generation types (prompt, variation, inpainting, outpainting) isn't working so well: forcing NotImplementedErrors and backwards workarounds like this. We can move this conversation to another venue, but should maybe consider making image generation drivers more granular.

Reply:
Can you please create a ticket to track this work? This feels like an important refactor pre-1.0.